### PR TITLE
Added methods required for future work in continuum mechanics

### LINKED ISSUE #16392: No function for calculating polar moment of inertia and section modulus in continuum_mechanics in beam.py
There is currently no function to calculate polar moment of inertia and section modulus of the beam . Both of these quantities are important in further future works in this module.

### PR DESCRIPTION
**Pull request by abhinavsagar**
#### References to other Issues or PRs

[Fixes #16392 ](https://github.com/sympy/sympy/issues/16392)

#### Brief description of what is fixed or changed
Added methods for calculation of polar moment of inertia, section modulus and torsional shear stress.

#### Release Notes

* physics.continuum_mechanics
  * added function for calculating polar moment of inertia.
  * added function for calculating section modulus.
  * added function for calculating torsional shear stress.

### CODE REVIEW

**Comment by abhinavsagar** [comment]
@moorepants Can you please review this. Although these methods are indexing from what is returned from previous methods,  I think all of these are important in future works in this module. I don't think unit tests are required for these . I will be more than happy to add tests or documentation for this as per the needs.

**Comment by oscargus** [comment]
Please fix the release notes to point at the correct module.

**Comment by sylee957** [comment]
https://github.com/sympy/sympy-bot/blob/master/sympy_bot/submodules.txt
Here are the valid module names for the release notes, I guess.
You should use `physics.continuum_mechanics`

File: sympy/physics/continuum_mechanics/beam.py [hunk 1]
```diff
@@ -1664,6 +1665,24 @@ def boundary_conditions(self):
         """
         return self._boundary_conditions
 
+    def diameter(self):
+        """
+        Returns Diameter of the beam object
+        """
+        return 1.13*math.sqrt(self.area)
```
Comment (sylee957) [inline sympy/physics/continuum_mechanics/beam.py]: Use `sqrt` from sympy, and also use symbolic coefficient.

**Comment by abhinavsagar** [comment]
@sylee957 @oscargus Thanks for mentioning. I have done the required changes.

**Comment by oscargus** [comment]
Btw, unit tests are always needed.

Regarding the comment from @sylee957, please use a symbolic coefficient. Although not really updated on the subject, I strongly doubt that 1.13 is the exact coefficient here.

File: sympy/physics/continuum_mechanics/beam.py [hunk 2]
```diff
@@ -1664,6 +1665,24 @@ def boundary_conditions(self):
         """
         return self._boundary_conditions
 
+    def diameter(self):
+        """
+        Returns Diameter of the beam object
+        """
+        return 1.13*sympy.sqrt(self.area)
```
Comment (moorepants) [inline sympy/physics/continuum_mechanics/beam.py]: Why would a beam always have a diameter? Beams can have any arbitrary cross sectional shape.

File: sympy/physics/continuum_mechanics/beam.py [hunk 3]
```diff
@@ -1664,6 +1665,24 @@ def boundary_conditions(self):
...
+
+    def polar_moment(self):
+        """
+        Returns the Polar moment of inertia of the beam object
+        """
+        return 2*self.second_moment
```
Comment (moorepants) [inline sympy/physics/continuum_mechanics/beam.py]: Shouldn't this be the sum of Ix and Iy?

File: sympy/physics/continuum_mechanics/beam.py [hunk 4]
```diff
@@ -1664,6 +1665,24 @@ def boundary_conditions(self):
...
+
+    def section_modulus(self):
+        """
+        Returns the Section modulus of the beam object
+        """
+        return (2*self.second_moment)/self.diameter()
```
Comment (moorepants) [inline sympy/physics/continuum_mechanics/beam.py]: Please point to a reference that defines this. I can't remember what it is.

**Comment by moorepants** [comment]
These may be good to add, but I recommend working on adding a more complete set of examples and unit tests for Beam3D. This class has little attention and is not really in a production quality state.

File: sympy/physics/continuum_mechanics/beam.py (hunk 4 above)
Comment (abhinavsagar) [reply sympy/physics/continuum_mechanics/beam.py]: @moorepants Check out this [link](https://en.wikipedia.org/wiki/Section_modulus)

**Comment by abhinavsagar** [comment]
@moorepants I understand your point. I have some other ideas and I will try to use these in my future works. And yeah I will try to generalize diameter and polar moment of inertia for all cases.

**Comment by moorepants** [comment]
I am closing this PR because Beam's can have any arbitrary cross sectional shape. They do not have diameters unless they happen to be circular cross sections and the section modulus is not a generalized method to handle any shape.

**Pull request closed without merging**